gspread/models.py

```diff
@@ -813,6 +813,96 @@ def findall(self, query):
         """
         return list(self._finder(filter, query))
 
+    def freeze(self, rows=None, cols=None):
+        """Freeze rows and/or columns on the worksheet.
+
+        :param rows: Number of rows to freeze.
+        :param cols: Number of columns to freeze.
+        """
+        grid_properties = {}
+
+        if rows is not None:
+            grid_properties['frozenRowCount'] = rows
+
+        if cols is not None:
+            grid_properties['frozenColumnCount'] = cols
+
+        if not grid_properties:
+            raise TypeError("Either 'rows' or 'cols' should be specified.")
+
+        fields = ','.join(
+            'gridProperties/%s' % p for p in grid_properties.keys()
+        )
+
+        body = {
+            'requests': [{
+                'updateSheetProperties': {
+                    'properties': {
+                        'sheetId': self.id,
+                        'gridProperties': grid_properties
+                    },
+                    'fields': fields
+                }
+            }]
+        }
+
+        return self.spreadsheet.batch_update(body)
+
+    @cast_to_a1_notation
+    def add_basic_filter(self, name=None):
```

Comment:
A basic filter also has (optionally) a SortSpec and a collection of FilterCriteria (optionally one criterion for each column included in the range). I think if we want to add basic filter support, we would want to support the full feature set of basic filters.

Reply:
Yeah, I thought about those but I figured I can sort and filter the data before pushing it... I find the basic filter most useful for allowing users of the sheet to do their own exploration.

However, that was purely thinking on my usecase... I think it makes sense to add the options here, but I'm not sure how to do it while still leveraging @cast_to_a1_notation. I'll think on it, but I'm thinking it might need some major re-work of @cast_to_a1_notation.

Reply:
Looking at these, they seem to be 'non-trivial' parameters. 

It's easy to implement their addition with no checking and letting the user build the correct JSON. An alternative is to create an object to represent each of those parameters so the user can build them programmatically and pass in the object. The latter implementation seems 'better' but it seems like too much effort for something that I'm not convinced will be used very often and I don't have the time to implement that.

However, I still think the addition of the basic functionality as proposed by this PR is quite useful to many (I had already implemented this in https://github.com/aiguofer/gspread-pandas and 'back-ported' it here for others to use).

If the simple param option (user passes in the dict) is good enough then I'm happy to implement it, otherwise I think this could be merged for now and the better implementation could be added later.